### PR TITLE
Add Identifiers container and case-sensitivity note

### LINKED ISSUE #319: Document case-sensitivity about names
### Related code PR

https://github.com/risingwavelabs/risingwave/issues/6051

### Which part(s) of the docs might be affected or should be updated? And how?

add a new page

### Reference

https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS:~:text=by%20changing%20the,are%20case%20insensitive

https://materialize.com/docs/sql/identifiers/

### PR DESCRIPTION
**Pull request by bernscode**
Add Identifiers container and case-sensitivity note.

## Info
- **Description**: 
Added Identifiers container and case-sensitivity note.

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/issues/6051

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/319

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/sql/sql-identifiers.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+* The first character of an identifier must be an ASCII letter (e.g., `a`-`z` and `A`-`Z`) or an underscore (`_`).
+* The remaining characters of an identifier must be ASCII letters (e.g., `a`-`z` and `A`-`Z`), underscores (`_`), ASCII digits (`0`-`9`), or dollar signs (`$`).
+* Non-ASCII characters in unquoted identifiers are not allowed.
+* You can circumvent any above rules by double-quoting the identifier (e.g., `"5_source"`). All characters inside a quoted identifier are taken literally, except double-quotes must be escaped by writing two adjacent double-quotes, as in (e.g., `"two""quotes"`).
+
+
+## Case sensitivity
+
+Identifiers are case-insensitive (e.g., `wave`, `WAVE`, and `wAve` are all the same). This can cause issues when column names come from data sources that do support case-sensitive names, such as Avro-formatted sources or CSV headers.
```
Comment (tabVersion) [inline docs/sql/sql-identifiers.md]: We should mention unquoted identifiers will be lowercase when  using them.

**Review by tabVersion — APPROVED** [review]
Rest look good to me

**Review by emile-00 — APPROVED** [review]

File: docs/sql/sql-identifiers.md (hunk 1 above)
Comment (hengm3467) [reply docs/sql/sql-identifiers.md]: Thanks. I added a bit more description about identifiers will be processes as in lower case internally.

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks. @bernscode

**Pull request merged**